taxi/views.py

```
@@ -111,7 +156,7 @@ class DriverDeleteView(LoginRequiredMixin, generic.DeleteView):
 def toggle_assign_to_car(request, pk):
     driver = Driver.objects.get(id=request.user.id)
     if (
-        Car.objects.get(id=pk) in driver.cars.all()
+            Car.objects.get(id=pk) in driver.cars.all()
```

This line performs two separate database queries: one `get` for the car and another to fetch all cars for the driver (`driver.cars.all()`). A more efficient and safer way to check for existence is using `.exists()`:

```python
if driver.cars.filter(id=pk).exists():
    driver.cars.remove(pk)
else:
    driver.cars.add(pk)
```

This approach avoids fetching unnecessary data and performs the check directly in the database. Also, `Car.objects.get(id=pk)` might raise a `DoesNotExist` exception if a car with the given `pk` is not found.